src/algebra/geom_sum.lean

```diff
@@ -367,3 +402,139 @@ begin
     ... = a + a/(b - 1)
         : by rw [mul_one, nat.add_mul_div_right _ _ (tsub_pos_of_lt hb), add_comm]
 end
+
+section order
+
+variables {n : ℕ} {x : α}
+
+lemma geom_sum_pos [ordered_semiring α] (hx : 0 < x) (hn : n ≠ 0) : 0 < geom_sum x n :=
+begin
+  induction n with k hk,
+  { exact (hn rfl).elim },
+  cases k,
+  { simp [@@zero_lt_one _ (nontrivial_of_lt _ _ hx)] },
+  rw [geom_sum_succ'],
+  exact add_pos (pow_pos hx _) (hk $ nat.succ_ne_zero k)
+end
+
+lemma geom_sum_pos_and_lt_one [ordered_ring α] (hx : x < 0) (hx' : 0 < x + 1) (hn : 1 < n) :
+  0 < geom_sum x n ∧ geom_sum x n < 1 :=
+begin
+  induction n with k hk,
+  { exact (lt_irrefl _ $ hn.trans zero_lt_one).elim },
+  rcases k with (_ | _ | k),
+  { exact (hn.ne rfl).elim },
```

Comment:
I think you can use `nat.exists_eq_add_of_lt` or whatever it's called to get rid of the cases `n = 0` and `n = 1`.

Reply:
I tried messing with this a little and it seemed to lead to more faff to me sadly